Makefile

```diff
@@ -45,11 +45,17 @@ ci-non-test: verify licenses vulncheck ## Runs checks other than tests
 ci: toolchain ci-non-tests ci-tests ## Run all steps used by continuous integration
 
 run: ## Run Karpenter controller binary against your local cluster
-	SYSTEM_NAMESPACE=${SYSTEM_NAMESPACE} go run ./cmd/controller/main.go \
-		--cluster-name=${CLUSTER_NAME} \
-		--cluster-endpoint=${CLUSTER_ENDPOINT} \
-		--aws-default-instance-profile=KarpenterNodeInstanceProfile-${CLUSTER_NAME} \
-		--leader-elect=false
+	kubectl create configmap -n ${SYSTEM_NAMESPACE} karpenter-global-settings \
+		--from-literal=aws.clusterName=${CLUSTER_NAME} \
+		--from-literal=aws.clusterEndpoint=${CLUSTER_ENDPOINT} \
+		--from-literal=aws.defaultInstanceProfile=KarpenterNodeInstanceProfile-${CLUSTER_NAME} \
+		--dry-run=client -o yaml | kubectl apply -f -
+
+	SYSTEM_NAMESPACE=${SYSTEM_NAMESPACE} KUBERNETES_MIN_VERSION="1.19.0-0" LEADER_ELECT=false \
+		go run ./cmd/controller/main.go
+
+clean-run: ## Clean resources deployed by the run target
+	kubectl delete configmap -n ${SYSTEM_NAMESPACE} karpenter-global-settings --ignore-not-found
```

Comment:
This feels really leaky. What are out other options rather than polluting the cluster?